lib/jekyll-admin/server.rb

```diff
@@ -75,7 +75,11 @@ def write_file(path, content)
       File.open(path, "wb") do |file|
         file.write(content)
       end
-      site.process
+      if ENV["RACK_ENV"]
+        site.process
+      else
+        site.read
+      end
```

Comment:
This isn't really a race condition in the traditional sense. We boot up a Watcher from `jekyll-watch` which runs `site.process` every time a file saved. The issue here appears to be: if `opts["watch"] && opts["serving"]` are passed to Jekyll, then the watcher in another thread runs `site.process` and you're also trying to run `site.process` at the same time and it can cause confusion. I think if there's another Jekyll process running, the call to `site.process` is not necessary here at all. Can you get away with just calling `site.read`?